test/index.js

```diff
@@ -69,88 +57,78 @@ describe('koa-body', () => {
         // return request contents to validate
         ctx.body = {
           _files: ctx.request.files, // files we populate
-          user: user // original request data
+          user, // original request data
         };
       })
-      .post('/echo_body', (ctx, next) => {
+      .post('/echo_body', (ctx) => {
         ctx.status = 200;
         ctx.body = ctx.request.body;
       })
-      .delete('/users/:user', (ctx, next) => {
-        const user = ctx.params.user;
+      .delete('/users/:user', (ctx) => {
+        const { user } = ctx.params;
         const multi = !!ctx.request.body.multi;
         if (multi) {
           database.users = database.users.filter(element => element.name !== user);
-        }
-        else {
+        } else {
           const index = database.users.findIndex(element => element === user);
           database.users.splice(index, 1);
         }
         ctx.status = 204;
         ctx.body = '';
       });
-    done();
   });
 
 
   /**
    * DEFAULTS - multipart: false
    */
-  it('should work with defaults - multipart: false, only `urlencoded` and `json` bodies',  (done) => {
+  it('should work with defaults - multipart: false, only `urlencoded` and `json` bodies', async () => {
     app.use(koaBody());
     app.use(router.routes());
 
-    request(http.createServer(app.callback()))
+    return request(http.createServer(app.callback()))
       .get('/users')
-      .expect(200, database)
-      .end( (err, res) => {
-        if (err) return done(err);
-        done();
-      });
+      .expect(200, database);
   });
 
 
   /**
    * MULTIPART - FIELDS
    */
-  it('should receive `multipart` requests - fields on .body object',  (done) => {
+  it('should receive `multipart` requests - fields on .body object', async () => {
     app.use(koaBody({ multipart: true }));
     app.use(router.routes());
 
-    request(http.createServer(app.callback()))
+    return request(http.createServer(app.callback()))
       .post('/users')
       .field('name', 'daryl')
       .field('followers', 30)
       .expect(201)
-      .end( (err, res) => {
-        if (err) return done(err);
-
+      .then((res) => {
```

Comment:
You've changed all these tests to async functions, but still use promise chaining. Could await the request, `const res = await request(...).expect(...)` and move the body of the then blocks out of the chain.

Reply:
good catch